### PR TITLE
[Merged by Bors] - feat: invariance of the derivative under translation

### PR DESCRIPTION
**Pull request by MichaelStollBayreuth**
This adds a new file `Mathlib.Analysis.Calculus.Deriv.Shift` that contains two lemmas
```lean
lemma HasDerivAt.comp_const_add {𝕜 : Type*} [NontriviallyNormedField 𝕜] (a x : 𝕜) {𝕜' : Type*}
    [NormedAddCommGroup 𝕜'] [NormedSpace 𝕜 𝕜'] {h : 𝕜 → 𝕜'} {h' : 𝕜'}
    (hh : HasDerivAt h h' (a + x)) :
    HasDerivAt (fun x ↦ h (a + x)) h' x := ...

lemma HasDerivAt.comp_add_const {𝕜 : Type*} [NontriviallyNormedField 𝕜] (x a : 𝕜) {𝕜' : Type*}
    [NormedAddCommGroup 𝕜'] [NormedSpace 𝕜 𝕜'] {h : 𝕜 → 𝕜'} {h' : 𝕜'}
    (hh : HasDerivAt h h' (x + a)) :
    HasDerivAt (fun x ↦ h (x + a)) h' x  := ...
```

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/Calculus/Deriv/Shift.lean [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+/-!
+### Invariance of the derivative under translation
+
+We show that if a function `h` has derivative `h'` at a point `a + x`, then `h (a + ·)`
+has derivative `h'` at `x`. Similarly for `x + a`.
+-/
+
+/-- Translation in the domain does not change the derivative. -/
+lemma HasDerivAt.comp_const_add {𝕜 : Type*} [NontriviallyNormedField 𝕜] (a x : 𝕜) {𝕜' : Type*}
+    [NontriviallyNormedField 𝕜'] [NormedAlgebra 𝕜 𝕜'] {h : 𝕜 → 𝕜'} {h' : 𝕜'}
```
Comment (urkud) [inline Mathlib/Analysis/Calculus/Deriv/Shift.lean:18]: You don't need the codomain to be a normed algebra. A normed space is enough.

Comment (MichaelStollBayreuth) [reply Mathlib/Analysis/Calculus/Deriv/Shift.lean]: OK, thanks; changed.

**Comment by MichaelStollBayreuth** [comment]
Done, thanks!

**Comment by urkud** [comment]
Why do you need `k'` to be a normed field at all? Can it be a normed group over `k`?
bors r-

**Comment by MichaelStollBayreuth** [comment]
It seems like `NormedAddCommGroup` is sufficient.

**Pull request closed without merging**